src/main/java/com/github/manolo8/darkbot/core/objects/gui/SizableGui.java

```diff
@@ -0,0 +1,13 @@
+package com.github.manolo8.darkbot.core.objects.gui;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.objects.Gui;
+
+public class SizableGui extends Gui {
+    public void update() {
+        if (address == 0) return;
+        super.update();
+        width = (int) Main.API.readDouble(address + 0x1F8);
+        height = (int) Main.API.readDouble(address + 0x200);
+    }
+}
```

Comment:
This SizableGui seems to be exactly what PopupGui is supposed to be, a generic superclass for guis that have the width and height in those addresses

Maybe the issue is PopupGui should not have the API.Singleton extension, that looks like an issue

Reply:
I not used PopupGui because it using incorrect way for getting isVisible for this case.